API-Samples/Java/app/src/main/java/apisamples/TokensStore.java

```diff
@@ -0,0 +1,134 @@
+import java.net.http.HttpRequest;
+import java.text.MessageFormat;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import static apisamples.HttpUtils.Utils.httpStringResponse;
+import static apisamples.VideoIndexerClient.*;
+
+public class TokensStore {
+
+    private final LoadingCache<String, TimedEntry> cache;
+    private static final String armAccessTokenKey = "armAccessToken";
+    private static final String viAccountTokenKey = "viAccountToken";
+    private static final long ExpiryBufferTimeInMs = TimeUnit.MINUTES.toMillis(5);
+
+    private final Gson gson;
+    private final ArmAccessTokenPermission permission;
+    private final ArmAccessTokenScope scope;
+
+
+    public TokensStore(ArmAccessTokenPermission permission, ArmAccessTokenScope scope) {
+        this.cache = Caffeine.newBuilder()
+                    .build(key -> new TimedEntry(key, 0));
+        this.gson = new GsonBuilder().setPrettyPrinting().create();
+        this.permission = permission;
+        this.scope = scope;
+
+        //Generate the First ARM and VI Access Token and place them in cache
+        this.getVIAccessToken();
+    }
+
+
+    /**
+     * TimedEntry represents an expiry enabled cache item to be placed in the Loading Cache lib
+     */
+    private class TimedEntry {
+        long expiryTime;
+        String value;
+
+        public TimedEntry(String value, long expiryTimeEpocMilli) {
+            this.value = value;
+            this.expiryTime = expiryTimeEpocMilli;
+        }
+        public boolean isExpired() {
+            return System.currentTimeMillis() > expiryTime;
+        }
+    }
+
+    public String getArmAccessToken() {
+        return this.getOrCreate(armAccessTokenKey, this::generateArmAccessToken);
+    }
+
+    public String getVIAccessToken() {
+        return this.getOrCreate(viAccountTokenKey, this::generateVIAccessToken);
+    }
+
+    public String getOrCreate(String key, Supplier<TimedEntry> valueSupplier) {
+        TimedEntry timedEntry = cache.getIfPresent(key);
+        if (timedEntry == null || timedEntry.isExpired()) {
+            timedEntry = valueSupplier.get();
+            cache.put(key, timedEntry);
+        }
+        return timedEntry.value;
+    }
+
+    /**
+     * Generate new Arm Access Token
+     * @return the Arm Access Token
+     */
+    private TimedEntry generateArmAccessToken() {
+
+        var tokenRequestContext = new TokenRequestContext();
+        tokenRequestContext.addScopes(String.format("%s/.default", AzureResourceManager));
+
+        DefaultAzureCredential defaultCredential = new DefaultAzureCredentialBuilder().build();
+        AccessToken accessToken = Objects.requireNonNull(defaultCredential.getToken(tokenRequestContext).block());
+        //Calculate Expiration token as Expiry value - Buffer Time ( default 5 Min)
+        long expiryEpocTime = accessToken.getExpiresAt().toInstant().toEpochMilli() - ExpiryBufferTimeInMs;

```

Comment:
consider doing the - 5m in the "isExpired"

Reply:
Like the idea but it can be missed by readers.